editor/src/components/canvas/context-menu-state.tsx

```diff
@@ -0,0 +1,47 @@
+import { atom, useAtom } from 'jotai'
+
+import { BodyMenuOpenClass } from '../../core/shared/utils'
+import React from 'react'
+
+export const ContextMenuStateAtom = atom([] as string[])
+export const ContextMenuOpen = atom((get) => {
+  const contextMenusOpen = get(ContextMenuStateAtom)
+  return contextMenusOpen.length > 0
+})
+
+export const useContextMenuState = () => {
+  const [values, setValues] = useAtom(ContextMenuStateAtom)
+
+  React.useEffect(() => {
+    if (values.length > 0) {
+      document.body.classList.add(BodyMenuOpenClass)
+    } else {
+      document.body.classList.remove(BodyMenuOpenClass)
+    }
+  }, [values])
```

Comment:
now that we have this atom, I think we should kill `BodyMenuOpenClass`. it can be done in a follow-up PR tho